src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (this.length === 0) {
+      return initialValue !== undefined ? initialValue : 'undefined';
```

Comment:
Returning 'undefined' as a string is not correct. You just can add a default value to the initialValue if you really need it
You can just remove this if statement, it's redundant